src/base.ts

```diff
@@ -309,7 +325,17 @@ Please check your network connection.`);
       throw error;
     }
   }
-
+  async promptPort(platform: string): Promise<number> {
+    const { port } = (await inquirer.prompt({
+      name: 'port',
+      type: 'input',
+      default: getDefaultPort(platform),
+      message: 'Enter the port your app listens to:',
+      validate: validatePort,
+    })) as { port: number };
+
+    return port;
+  }
```

Comment:
I noticed that the `promptPort` method is repeated in `deploy.ts` as well. Additionally, `base.ts` doesn’t seem like the appropriate place for a utility function like this.

Why not move `promptPort` to the `utils` directory? This would allow us to reuse it wherever needed and keep `base.ts` focused and simple.

Keeping `base.ts` as clean as possible is important for maintainability and readability.

Reply:
I tried to follow the approach that has been used so far. there is also promptNetwork() and promptProject() that is defined in base.ts. I suggest that we let them be in base.ts for now and and move all of them later.